docs/reference/item.md

```diff
@@ -177,7 +177,40 @@ item of the reviewed attribute, not the parent item of the link).  The
 fingerprint of the link is does **not** contribute to the fingerprint of the
 item.
 
-## `ref`
+## `references` (new array behavior)
+
+An array of external references. An item may reference a number of external
+files. The external references are displayed in a published document.
+
+Doorstop will search in the project root for a file matching the specified
+reference. If multiple matching files exist, the first found will be used.
+
+The value of this attribute contributes to the [fingerprint](item.md#reviewed)
+of the item.
+
+### Example: Reference files
+
+```yaml
+references:
+- path: tests/test1.cpp
+  type: file
+- path: tests/test2.cpp
+  type: file
+```
+
+### Note: new behavior vs old behavior
+
+Note: Newer `references` attribute is a new array behavior compared to the 
+original `ref` attribute's string behavior. The old behavior supports referencing
+only one file via file name or referencing a file that contains a given
+keyword.
+
+The new behavior of `references` attribute allows referencing many files. It
```

Comment:
@jacebrowning @sebhub 

Is it ok to keep the scope of this PR as it is without implementing the `keyword` to also support the old `ref` behavior?

I am fine with both implementing it here or doing a separate PR. In both cases, the Item has to be refactored because it is now again close to being 1000 lines. 

Having the above said, I think the PR is ready for review in its current shape. Please take a look.